src/GoogleCloudPubSub/GoogleCloudPubSub.ts

```diff
@@ -127,9 +127,12 @@ export class GoogleCloudPubSub implements GCPubSub {
 
     subscription.on('message', (message: Message): void => {
       const extendedMessage: ExtendedMessage<T> = new ExtendedMessage<T>(message);
+      const logMessage: ExtendedMessage<T> = new ExtendedMessage<T>(message);
```

Comment:
instead of creating a new instance of `ExtendedMessage`, you should omit the `originalMessage` property; something like:

```suggestion
      const logMessage: ExtendedMessage<T> = { ...extendedMessage, originalMessage: '' };
```

Not the cleanest solution, there are a few others 😄

Reply:
Thanks, it's the only way working for me. I've tried many ways but haven't succeeded. The fact that `originalMessage` is defined as "obligatory private readonly" property in ExtendedMessage makes it difficult to remove it from the object. 

The object couldn't not detect for deleting or destructuring that property. But when we spreads it out to print the log, we will still see it visible. I think that's a drawback of TypeScript `private` keyword because it's not a real privacy. ECMAScript now has a new feature `#` for private property. I think it's more private because if I spread the objects out, the private property won't be included.

Reply:
>  I think that's a drawback of TypeScript private keyword because it's not a real privacy.

Yes I see your point. 

> The object couldn't not detect for deleting or destructuring that property. But when we spreads it out to print the log, we will still see it visible.

Yes, but you can still override a private property. That's why I suggest you override in another variable the private property (not clean I agree, but lighter than creating a new instance).

```
class A {
    public a;
    public b;
    private c;

    constructor(params: { a: number; b: number; c: string }) {
        this.a = params.a;
        this.b = params.b;
        this.c = para...
```

[Playground Link](https://www.typescriptlang.org/play?module=5#code/MYGwhgzhAECC0G8CwAoa7oAcCuAjEAlsNGANyoZZ6HG7loaYBOBAbmAC4Cm0w9FGYAHsAdhA5NswDkKYAKTGCZgAthABciEppHYVuLk1LRcOvQaO9N4liIDmAAoC+ASkQDK6DgAsCEAHRg0AC8WEqqAWQenj5+-rghYcpq8fSeGLEBxKGKyVlpGE6o0XZcHNClHADCcm7IDOlMZdhMItCZ-nzRRSg9qMJi5SJcAO7wocMjcHIIYJoAjAA0JpoATMvAmgDkAMxbrsb9ohBCIFz+IEJ2cpOwLsYA9A-QACZ+mOAAntBHYqfnl2uCGg-lBtw22yErEMLBeXC2znu0Cer3eX2gQA)